contracts/core/SequencerSoloClient.sol

```diff
@@ -75,7 +77,9 @@ contract SequencerSoloClient is ILightClient {
     }
 
     /**
-     * @inheritdoc ILightClient
+     * @dev Checks if the current trusted optimistic consensus state
+     * can be used to perform the membership test and if so, verifies the proof
+     * @dev reverts if the proof is not valid (i.e. if the key is not included in the proof)
      */
     function verifyMembership(Ics23Proof calldata proof, bytes calldata key, bytes calldata expectedValue)
```

Comment:
Did we find out what the reason for using AppHash at n-1 for proof at n is?

```
// a proof generated at height H can only be verified against state root (app hash) from block H - 1.
// this means the relayer must have updated the contract with the app hash from the previous block and
// that is why we use proof.height - 1 here.
```

Reply:
we didn't go super down into the weeds, but I think it was something to do with how ics23 proofs work, and that the relayer was assuming to use `height - 1` so it was still consistent with what the relayer expects

Reply:
I believe this is because you can only prove things had already happened and are in the app state which isn't true until you've completed the block and are at the next height. i.e A proof at height, H, uses the merkle tree of app state built a height, H - 1